File: src/plugin/command/administrator/PerformAnimationCommandPlugin.java
package plugin.command.administrator;

import com.rs.game.entity.actor.mask.Animation;
import com.rs.game.entity.actor.player.Player;
import com.rs.game.plugin.type.CommandPlugin;
import plugin.command.CommandManifest;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/5/2017
 */
@CommandManifest(description = "Performs an animation", types = { Integer.class })
public class PerformAnimationCommandPlugin extends CommandPlugin {
	
	@Override
	public void handle(Player player, String[] args, boolean console, boolean clientCommand) {
		int id = intParam(args, 1);
		int speed = intParamOrDefault(args, 2, 0);
		player.setNextAnimation(new Animation(id, speed));
	}
	
	@Override
	public String[] identifiers() {
		return arguments("anim");
	}
}

File: src/main/kotlin/plugin/command/administrator/PerformAnimationCommandPlugin.kt
package plugin.command.administrator

import org.redrune.game.content.plugin.type.CommandPlugin
import org.redrune.game.entity.actor.mask.Animation
import org.redrune.game.entity.actor.player.Player
import plugin.command.CommandManifest

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 9/5/2017
 */
@CommandManifest(description = "Performs an animation", types = [Int::class])
class PerformAnimationCommandPlugin : CommandPlugin() {
    override fun handle(player: Player, args: Array<String>, console: Boolean, clientCommand: Boolean) {
        val id = intParam(args, 1)
        val speed = intParamOrDefault(args, 2, 0)
        player.nextAnimation = Animation(id, speed)
    }

    override fun identifiers(): Array<String> {
        return arguments("anim")
    }
}